ocaml/xapi/message_forwarding.ml

```diff
@@ -1002,6 +1002,8 @@ functor
               debug "Pool.enable_tls_verification sleeping on fistpoint" ;
               Thread.delay 5.0
             done ;
+            Db.Cluster.get_all ~__context
+            |> List.iter (Xapi_cluster_helpers.Pem.maybe_write_new ~__context) ;
```

Comment:
Do we need to tell each one to write it? Or can we tell it to just the local clusterd, and let the local clusterd run distribute state across the cluster?

Reply:
The way it works is that xapi finds an enabled cluster host, gives it a pem blob, and it's that cluster host's job to tell the rest of the cluster hosts (with `distribute_state`). Later on xapi tells each cluster daemon to restart, and at that point they will be using the new pem.

If this distribute state call fails to update one of the cluster hosts, I am envisioning that an admin would (a) emergency disable verification on that host (b) fix the network problem (c) disable+(re)enable the cluster host